exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/trace_exporter/__init__.py

```diff
@@ -236,29 +236,42 @@ def _translate_data(
         sdk_resource_instrumentation_library_spans = {}
 
         for sdk_span in data:
-
-            if sdk_span.resource not in (
-                sdk_resource_instrumentation_library_spans.keys()
-            ):
+            instrumentation_library_spans_map = (
+                sdk_resource_instrumentation_library_spans.get(
+                    sdk_span.resource, {}
+                )
+            )
+            # If we haven't seen the Resource yet, add it to the map
+            if not instrumentation_library_spans_map:
+                sdk_resource_instrumentation_library_spans[
+                    sdk_span.resource
+                ] = instrumentation_library_spans_map
+            instrumentation_library_spans = (
+                instrumentation_library_spans_map.get(
+                    sdk_span.instrumentation_info
+                )
+            )
+            # If we haven't seen the InstrumentationInfo for this Resource yet, add it to the map
+            if not instrumentation_library_spans:
                 if sdk_span.instrumentation_info is not None:
-                    instrumentation_library_spans = (
-                        InstrumentationLibrarySpans(
-                            instrumentation_library=InstrumentationLibrary(
-                                name=sdk_span.instrumentation_info.name,
-                                version=sdk_span.instrumentation_info.version,
-                            )
+                    instrumentation_library_spans_map[
+                        sdk_span.instrumentation_info
+                    ] = InstrumentationLibrarySpans(
+                        instrumentation_library=InstrumentationLibrary(
+                            name=sdk_span.instrumentation_info.name,
+                            version=sdk_span.instrumentation_info.version,
                         )
                     )
-
                 else:
-                    instrumentation_library_spans = (
-                        InstrumentationLibrarySpans()
-                    )
-
-                sdk_resource_instrumentation_library_spans[
-                    sdk_span.resource
-                ] = instrumentation_library_spans
-
+                    # If no InstrumentationInfo, store in None key
+                    instrumentation_library_spans_map[
+                        sdk_span.instrumentation_info
```

Comment:
I see why this is assigning the key here, but it was confusing to me that the same key was used in both cases.

Reply:
Addressed privately.